modules/system/assets/js/snowboard/utilities/Debounce.js

```diff
@@ -0,0 +1,18 @@
+export default (fn) => {
+    // This holds the requestAnimationFrame reference, so we can cancel it if we wish
+    let frame;
+
+    // The debounce function returns a new function that can receive a variable number of arguments
+    return (...params) => {
+        // If the frame variable has been defined, clear it now, and queue for next frame
+        if (frame) {
+            cancelAnimationFrame(frame);
+        }
+
+        // Queue our function call for the next frame
+        frame = requestAnimationFrame(() => {
+            // Call our function and pass any params we received
+            fn(...params);
+        });
+    };
+};
```

Comment:
What does this plugin actually do / is it actually being used? I couldn't find any references to it beyond where it gets registered and added.

Reply:
I thought it was used by the Transition utlity, but apparently not (although it should).

This utility is synonymous with the Vue's `nextTick` feature - essentially allowing you to create a callback that will fire on the next "frame". It's useful for things like event handling and DOM manipulation where you need to ensure that everything in the previous frame is completed before running the callback.

For example, in the Transition library, a particular class is available for one frame to ensure that CSS transition is correctly initialised. We would use a debounce to remove the class in the next frame so the transition can take effect. This is similar to Vue transitions.

Reply:
Gotcha, if you could update the transition to use it and then add some docs to the Utilities page that I added earlier today for it that would be great! Finish line is in sight, I'm pretty excited for this to be released!